tests/unit-tests/test_decoder.py

```diff
@@ -981,17 +982,62 @@ def test_MLPEFeatEdgeDecoder(h_dim, feat_dim, out_dim, num_ffn_layers):
         prediction = decoder.predict(g, encoder_feat, efeat)
         pred = out.argmax(dim=1)
         assert_almost_equal(prediction.cpu().numpy(), pred.cpu().numpy())
+
 @pytest.mark.parametrize("in_dim", [16, 64])
 @pytest.mark.parametrize("out_dim", [1, 8])
 def test_EntityRegression(in_dim, out_dim):
-    decoder = EntityRegression(h_dim=in_dim)
+    decoder = EntityRegression(h_dim=in_dim, use_bias=False)
     assert decoder.in_dims == in_dim
     assert decoder.out_dims == 1
+    assert not hasattr(decoder, 'bias')
 
-    decoder = EntityRegression(h_dim=in_dim, out_dim=out_dim)
+    decoder = EntityRegression(h_dim=in_dim, out_dim=out_dim, use_bias=True)
     assert decoder.in_dims == in_dim
     assert decoder.out_dims == out_dim
 
+    decoder.eval()
+
+    in_tensor = th.ones((1,in_dim))
+    with th.no_grad():
+        th.nn.init.eye_(decoder.decoder)
+        out = decoder.predict(in_tensor)
+    assert out.shape == (1, out_dim)
+    assert th.all(out == 1)
+
+    with th.no_grad():
+        th.nn.init.constant_(decoder.bias, 5)
+        out = decoder.predict(in_tensor)
+    assert out.shape == (1, out_dim)
+    assert th.all(out == 6)
+
+
+@pytest.mark.parametrize("in_dim", [16, 64])
+@pytest.mark.parametrize("num_classes", [4, 8])
+def test_EntityClassifier(in_dim, num_classes):
+
+    decoder = EntityClassifier(in_dim=in_dim, num_classes=num_classes, multilabel=False, use_bias=False)
+    assert decoder.in_dims == in_dim
+    assert decoder.out_dims == num_classes
+    assert not hasattr(decoder, 'bias')
+
+    decoder = EntityClassifier(in_dim=in_dim, num_classes=num_classes, multilabel=False, use_bias=True)
+    assert decoder.in_dims == in_dim
+    assert decoder.out_dims == num_classes
+
+    decoder.eval()
+
+    in_tensor = th.ones(1, in_dim)
+    with th.no_grad():
+        th.nn.init.eye_(decoder.decoder)
+        decoder.decoder.data[0][0] += 1
```

Comment:
What is the purpose of this assignment?
It seems if the `in_tensor` is `th.ones(1, in_dim)` and using `.init.eye_`, the results of `decoder.predict(in_tensor)` will always be 0 no matter `+=1` or not.

In addition, it is very hard to see the logic of this test case. My guess is that by adding a positive value in the `.data[0][N]` dimension, the predicted class will be the `N`. If this is what you want to test `EntityClassifier`, it is better to use a variable to control, showing this logic explicitly. And adding some comment lines will help too.